### PR TITLE
Expose origin auth state and errors to squirrel

### PR DESCRIPTION
**Pull request by ASpoonPlaysGames**
Also moves NSIsMasterServerAuthenticated out of scriptserverbrowser.cpp because it didn't really fit there

This will be used for showing failed origin auth errors in the game's UI

### CODE REVIEW

**Review by pg9182 — APPROVED** [review]
Looks good, not tested.

File: NorthstarDLL/CMakeLists.txt [hunk 1]
```diff
@@ -85,6 +85,7 @@ add_library(NorthstarDLL SHARED
             "scripts/client/scriptbrowserhooks.cpp"
             "scripts/client/scriptmainmenupromos.cpp"
             "scripts/client/scriptmodmenu.cpp"
+			"scripts/client/scriptoriginauth.cpp"
```
Comment (F1F7Y) [inline NorthstarDLL/CMakeLists.txt:88]: Tabs

**Review by F1F7Y — CHANGES_REQUESTED** [review]

File: NorthstarDLL/CMakeLists.txt (hunk 1 above)
Comment (ASpoonPlaysGames) [reply NorthstarDLL/CMakeLists.txt]: Resolved in [f16c11d](https://github.com/R2Northstar/NorthstarLauncher/pull/468/commits/f16c11db8e86f7f46a9cd1886840105d8bbb1dfe)

**Review by F1F7Y — APPROVED** [review]
Code looks good

**Review by Alystrasz — APPROVED** [review]
This PR effectively exposes information to the Squirrel VM.
Confirmed working in testing.

**Comment by GeckoEidechse** [comment]
Merging based on previous reviews. Didn't look at this much myself.

**Pull request merged**